### ISSUE TITLE
Request textDocument/documentSymbol failed when typing calss

### ISSUE THREAD

**Issue by Igorgro** [dotnet/vscode-csharp#5891]
## Environment data
`dotnet --info` output:
```
.NET SDK:
 Version:   7.0.302
 Commit:    990cf98a27

Runtime Environment:
 OS Name:     arch
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.302/

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  6.0.408 [/usr/share/dotnet/sdk]
  7.0.302 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.16 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.16 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```
VS Code version: 1.79.2
C# Extension version: v2.0.248

## ~~OmniSharp~~ .NET host log

```
[Error - 20:20:55] Request textDocument/documentSymbol failed.
Error: name must not be falsy
	at k.validate (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:97:26587)
	at new k (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:97:26882)
	at V (/home/igormahov/.vscode/extensions/ms-dotnettools.csharp-2.0.248-linux-x64/dist/extension.js:2:1969276)
	at c (/home/igormahov/.vscode/extensions/ms-dotnettools.csharp-2.0.248-linux-x64/dist/extension.js:2:2007332)
	at t.map (/home/igormahov/.vscode/extensions/ms-dotnettools.csharp-2.0.248-linux-x64/dist/extension.js:2:2007418)
	at Object.asDocumentSymbols (/home/igormahov/.vscode/extensions/ms-dotnettools.csharp-2.0.248-linux-x64/dist/extension.js:2:1977094)
	at r (/home/igormahov/.vscode/extensions/ms-dotnettools.csharp-2.0.248-linux-x64/dist/extension.js:2:1915643)
	at runNextTicks (node:internal/process/task_queues:61:5)
	at process.processImmediate (node:internal/timers:437:9)
	at async a.provideDocumentSymbols (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:101:37935)
```

## Steps to reproduce

1. Install latest vscode and preview version of C# extension
2. Create .net/c# project
3. Create new .cs file
4. Type the following code:
    ```csharp
    namespace Demo;
    public class
    ```
5. After typing last letter 's' in word 'class', the error 'Request textDocument/documentSymbol failed.' appears.

## Expected  behavior
No errors are shown

## Actual behavior
C# extension throws an error

## Additional context
--

**Comment by arkalyanms** [comment]
@richlander I believe hit this too.

**Comment by rledyard** [comment]
I have the same issue, but under different circumstances. For me, this same error occurs when inserting new code into an existing line via a custom VS Code snippet. I type the short-hand for the snippet, hit tab, and have the following error occur:
[Error - 9:12:11 AM] [LanguageServerHost]System.ArgumentException: argumentCount < argumentIndex. 2 < 4 (Parameter 'argumentIndex')
   at Microsoft.CodeAnalysis.SignatureHelp.SignatureHelpItems..ctor(IList`1 items, TextSpan applicableSpan, Int32 argumentIndex, Int32 argumentCount, String argumentName, Nullable`1 selectedItem) in /_/src/Features/Core/Portable/SignatureHelp/SignatureHelpItems.cs:line 76
   at Microsoft.CodeAnalysis.SignatureHelp.AbstractSignatureHelpProvider.CreateSignatureHelpItems(IList`1 items, TextSpan applicableSpan, Nullable`1 state, Nullable`1 selectedItemIndex, Int32 parameterIndexOverride) in /_/src/Features/Core/Portable/SignatureHelp/AbstractSignatureHelpProvider.cs:line 68
   at Microsoft.CodeAnalysis.CSharp.SignatureHelp.InvocationExpressionSignatureHelpProviderBase.GetItemsWorkerAsync(Document document, Int32 position, SignatureHelpTriggerInfo triggerInfo, SignatureHelpOptions options, CancellationToken cancellationToken) in /_/src/Features/CSharp/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.cs:line 126
   at Microsoft.CodeAnalysis.SignatureHelp.AbstractSignatureHelpProvider.GetItemsAsync(Document document, Int32 position, SignatureHelpTriggerInfo triggerInfo, SignatureHelpOptions options, CancellationToken cancellationToken) in /_/src/Features/Core/Portable/SignatureHelp/AbstractSignatureHelpProvider.cs:line 302
   at Microsoft.CodeAnalysis.LanguageServer.Handler.SignatureHelpHandler.HandleRequestAsync(TextDocumentPositionParams request, RequestContext context, CancellationToken cancellationToken) in /_/src/Features/LanguageServer/Protocol/Handler/SignatureHelp/SignatureHelpHandler.cs:line 58
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`3.StartRequestAsync(TRequestContext context, CancellationToken cancellationToken) in /_/src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs:line 176
[Error - 9:12:11 AM] Request textDocument/signatureHelp failed.
  Message: argumentCount < argumentIndex. 2 < 4 (Parameter 'argumentIndex')
  Code: -32000 
[object Object]

I'm using version 2.0.248 of the C# extension - the C# project is pinned to SDK 6.0.411 by the global.json file for the solution.

**Comment by beccamc** [comment]
@rledyard This looks like a different issue to me. The error shown is `Request textDocument/signatureHelp failed`, right? 

Do you mind filing a new bug so we can track this problem? Thanks!

**Comment by beccamc** [comment]
This is fixed in the latest version 2.0.283.  Please check it out and let us know if you see anything else! 



![Image](https://github.com/dotnet/vscode-csharp/assets/9122518/59cda195-773d-4f5e-9b93-b35aa96586bc)
